### ISSUE TITLE
Allow configuring Go build tags to go-build and go-test checkers

### ISSUE THREAD

**Issue by sqs** [flycheck/flycheck#503]
[Go build tags](http://golang.org/pkg/go/build/) are comments at the top of Go files that define constraints that must be satisfied for the file to be included in a build. Go build tags are used to provide different implementations for different OSes and architectures and to control whether other kinds of features are included in builds.

For example, consider this Go file:

```go
// +build foo

package bar
```

It is ignored by `go build` because the `foo` build tag is not satisfied. If you call `go build -tags foo`, however, the file will be included in the build.

It is not possible to specify the build tags that the [go-build](https://github.com/flycheck/flycheck/blob/76b1964609c01e2b2b6121bdfad3450f0bdb56c0/flycheck.el#L5223) and [go-test](https://github.com/flycheck/flycheck/blob/76b1964609c01e2b2b6121bdfad3450f0bdb56c0/flycheck.el#L5242) checkers use. This means that when you edit a file with unsatisfied build tags, Flycheck doesn't check it.

There should be a configuration setting for the Go build tags you want to use. This is only necessary for checkers that actually build code; AST/parser-only checkers such as `gofmt`, `go-errcheck`, and `golint` operate on files, not packages, so their behavior is not affected by build tags.

You can work around this issue by setting `go-build` and `go-test` to call a shell script wrapper that adds `-tags whatever`, but that's clunky and requires some non-trivial parsing of the arguments.

I can submit a PR but wanted to post this issue first in case there were any other comments.

**Comment by drewwells** [comment]
This would be a terrific improvement. Another missing option is `-i` which installs dependent packages during build.

**Comment by lunaryorn** [comment]
Closing as duplicate.
